src/test/java/org/opensearch/security/SlowIntegrationTests.java

```diff
@@ -210,4 +217,38 @@ public void testNodeClientDisallowedWithNonServerCertificate2() throws Exception
             Assert.fail(e.toString());
         }
     }
+
+    @Test
+    public void testDelayInSecurityIndexInitialization() throws Exception {
+        final Settings settings = Settings.builder()
+            .put(ConfigConstants.SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true)
+            .put("cluster.routing.allocation.exclude._ip", "127.0.0.1")
+            .build();
+        try {
+            setup(Settings.EMPTY, null, settings, false);
+            clusterHelper.nodeClient()
+                .admin()
+                .indices()
+                .create(new CreateIndexRequest("test-index").timeout(TimeValue.timeValueSeconds(10)))
+                .actionGet();
+            clusterHelper.waitForCluster(ClusterHealthStatus.GREEN, TimeValue.timeValueSeconds(5), ClusterConfiguration.DEFAULT.getNodes());
+            Assert.fail("Expected IOException here due to red cluster state");
```

Comment:
Rewrite with `assertThrows(...)`

Reply:
refactored to use assertThrows. Its a lot cleaner with assertThrows!